src/modules/share/share.controller.ts

```diff
@@ -0,0 +1,219 @@
+      user = await this.userUseCases.getUserByUsername(user.email);
+    }
+    const share = await this.shareUseCases.getShareByToken(token, user);
+    // notify no analytics if not folder
+    return share.toJSON();
+  }
+
+  @Put('/:shareId')
+  @HttpCode(200)
+  @ApiOperation({
+    summary: 'Update share by id',
+  })
+  @ApiOkResponse({ description: 'Get share updated' })
+  @Public()
+  async updateShareByToken(
+    @User() user: any,
+    @Param('shareId') shareId: string,
+    @Body() content: UpdateShareDto,
+  ) {
+    if (!user.id && user.email) {
+      user = await this.userUseCases.getUserByUsername(user.email);
+    }
+    const share = await this.shareUseCases.updateShareById(
+      parseInt(shareId),
+      user,
+      content,
+    );
+    // notify no analytics if not folder
+    return share;
+  }
+
+  @Post('file/:fileId')
+  @ApiOperation({
+    summary: 'Generate Shared Token by file Id',
+  })
+  @ApiOkResponse({ description: 'Get Token of share' })
+  async generateSharedTokenToFile(
+    @User() user: any,
+    @Param('fileId') fileId: string,
+    @Body() body: CreateShareDto,
+    @Res() res: Response,
+  ) {
+    const share = await this.shareUseCases.createShareFile(fileId, user, body);
+
+    res.status(share.created ? HttpStatus.CREATED : HttpStatus.OK).json({
+      created: share.created,
+      token: share.item.token,
+    });
+  }
+
+  @Post('folder/:folderId')
+  @HttpCode(200)
+  @ApiOperation({
+    summary: 'Generate Shared Token by folder Id',
+  })
+  @ApiOkResponse({ description: 'Get token of share' })
+  async generateSharedTokenToFolder(
+    @User() user: any,
+    @Param('folderId') folderId: string,
+    @Body() body: CreateShareDto,
+    @Res() res: Response,
+  ) {
+    const share = await this.shareUseCases.createShareFolder(
+      parseInt(folderId),
+      user,
+      body,
+    );
+
+    res.status(share.created ? HttpStatus.CREATED : HttpStatus.OK).json({
+      created: share.created,
+      token: share.item.token,
+    });
+  }
+
+  @Get('down/files')
+  @HttpCode(200)
+  @ApiOperation({
+    summary: 'Get all files by token paginated',
+  })
+  @ApiOkResponse({ description: 'Get all files' })
+  @Public()
+  async getDownFiles(@User() user: any, @Query() query: GetDownFilesDto) {
+    const { token, folderId, code, page, perPage } = query;
+    if (!user.id && user.email) {
+      user = await this.userUseCases.getUserByUsername(user.email);
+    }
+    const share = await this.shareUseCases.getShareByToken(token, user);
+    share.decryptMnemonic(code);
+    const network = await this.userUseCases.getNetworkByUserId(
+      user.id,
+      share.mnemonic,
+    );
+    const files = await this.fileUseCases.getByFolderAndUser(
+      folderId,
+      user,
+      false,
+      parseInt(page),
+      parseInt(perPage),
+    );
+
+    for (const file of files) {
+      file.encryptionKey =
+        await this.fileUseCases.getEncryptionKeyFileFromShare(
```

Comment:
Consider doing this in bulk and then, iterate over the result to set the `encryptionKey` for each file